src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/SampleNameMap.java

```diff
@@ -0,0 +1,274 @@
+                if ( split.length == 3 ) {
+                    vcfIndexPath = split[2].trim();
+
+                    if ( vcfIndexPath.isEmpty() ) {
+                        throw new UserException.BadInput("Found a line in the sample name map file with an empty or all-whitespace value for the index:\n" + "\"" + line + "\"");
+                    }
+                }
+
+                try {
+                    final URI existingVCFPath = sampleNameToVcfPath.put(sample, new URI(vcfPath));
+                    if (existingVCFPath != null){
+                        throw new UserException.BadInput("Found two mappings for the same sample: " + sample + "\n" + vcfPath + "\n" + existingVCFPath);
+                    }
+
+                    if ( vcfIndexPath != null ) {
+                        final URI existingVCFIndexPath = sampleNameToVcfIndexPath.put(sample, new URI(vcfIndexPath));
+                        if (existingVCFIndexPath != null) {
+                            throw new UserException.BadInput("Found two indices for the same sample: " + sample + "\n" + vcfIndexPath + "\n" + existingVCFIndexPath);
+                        }
+                    }
+
+                    if (checkVcfIsCompressedAndIndexed) {
+                        GATKGenomicsDBUtils.assertVariantFileIsCompressedAndIndexed(IOUtils.getPath(vcfPath), vcfIndexPath == null ? null : IOUtils.getPath(vcfIndexPath));
+                    }
+                }
+                catch(final URISyntaxException e) {
+                    throw new UserException("Malformed URI: " + e.toString());
+                }
+            }
+        } catch (final IOException e) {
+            throw new UserException.CouldNotReadInputFile(sampleToFileMapPath, "exception while reading sample->filename mapping file",  e);
+        }
+    }
+
+    /**
+     * Tests whether the sample name is legal. Sample names must be non-empty, and
+     * may have internal whitespace but not leading/trailing whitespace.
+     *
+     * @param sampleName sample name to test
+     * @return true if sampleName is legal, otherwise false
+     */
+    private boolean sampleNameIsLegal(final String sampleName) {
+        return sampleName != null &&
+                ! sampleName.trim().isEmpty() &&
+                sampleName.trim().equals(sampleName);
+    }
+
+    /**
+     * Add a new sample mapping
+     *
+     * @param sampleName name of the sample
+     * @param vcfPath path to the VCF for the sample
+     */
+    public void addSample(final String sampleName, final URI vcfPath) {
+        addSample(sampleName, vcfPath, null);
+    }
+
+    /**
+     * Add a new sample mapping
+     *
+     * @param sampleName name of the sample
+     * @param vcfPath path to the VCF for the sample (not null)
+     * @param vcfIndexPath path to the index for the sample (may be null)
+     */
+    public void addSample(final String sampleName, final URI vcfPath, final URI vcfIndexPath) {
```

Comment:
Should the loadSampleMap file call addSample for each line instead of duplicating the checks?